src/decoding/object.rs

```diff
@@ -90,6 +90,28 @@ impl<'obj, 'ser: 'obj> Object<'obj, 'ser> {
         }
     }
 
+    /// Try to treat the object as a byte string, mapping [`Object::Bytes(v)`] into
+    /// [`Ok(v)`]. Any other variant results in an [`Error::UnexpectedElement`].
+    ///
+    /// [`Object::Bytes(v)`]: self::Object::Bytes
+    /// [`Ok(v)`]: https://doc.rust-lang.org/std/result/enum.Result.html#variant.Ok
+    /// [`Error::UnexpectedElement`]: self::Error::UnexpectedElement
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bendy::decoding::{Error, Object};
+    ///
+    /// let x = Object::Bytes(b"foo");
+    /// assert_eq!(b"foo", x.try_into_bytes().unwrap());
+    ///
+    /// let x = Object::Integer("foo");
+    /// assert!(x.try_into_bytes().is_err());
+    /// ```
+    pub fn try_into_bytes(self) -> Result<&'ser [u8], Error> {
```

Comment:
This doesn't necessarily belong in this changeset, but does it make sense to include `self` in the error, so that if decoding as one type fails, users can try again as another type? Of course, that enhanced error type would then be `Into<Error>`, so `?` would continue to work as expected

Reply:
As discussed this will be postponed until we or someone else encounters a real use case.